cloud/amazon/ec2_cert

```diff
@@ -0,0 +1,219 @@
+    msg: "{{ cert }}"
+'''
+
+import sys
+import time
+
+try:
+    import boto.iam
+except ImportError:
+    print "failed=True msg='boto required for this module'"
+    sys.exit(1)
+
+AWS_REGIONS = list(map(lambda r: r.name,
+                       boto.iam.regions()))
+
+def list_certs (iam):
+    data = iam.list_server_certs()
+    response = data.list_server_certificates_response
+    result = response.list_server_certificates_result
+    return result.server_certificate_metadata_list
+
+def get_server_cert(iam, cert_name):
+    certs = list_certs(iam)
+    matching_certs = (c for c in certs if c.server_certificate_name == cert_name)
+    return next(matching_certs, None)
+
+def upload_cert (iam,
+                 cert_name,
+                 cert_public_key,
+                 cert_private_key,
+                 cert_chain,
+                 cert_path):
+
+    cert = get_server_cert(iam, cert_name)
+
+    if cert:
+        changed = False
+    else:
+        changed = True
+        data = iam.upload_server_cert(cert_name,
+                                      cert_public_key,
+                                      cert_private_key,
+                                      cert_chain,
+                                      cert_path)
+        response = data.upload_server_certificate_response
+        result = response.upload_server_certificate_result
+        cert = result.server_certificate_metadata
+
+    return {'changed': changed, 'cert': cert}
+
+def delete_cert (iam, cert_name):
+    if get_server_cert(iam, cert_name):
+        changed = True
+        iam.delete_server_cert(cert_name)
+    else:
+        changed = False
+
+    return {'changed': changed, 'cert': None}
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            region = dict(aliases=['aws_region', 'ec2_region'], choices=AWS_REGIONS, required=True),
+            aws_secret_key = dict(aliases=['ec2_secret_key', 'secret_key'], no_log=True),
+            aws_access_key = dict(aliases=['ec2_access_key', 'access_key']),
+            cert_name = dict(),
+            cert_public_key = dict(),
+            cert_private_key = dict(no_log=True),
+            cert_chain = dict(),
+            cert_path = dict(default="/"),
+            state = dict(default="present", choices=['present', 'absent', 'list'])
+        )
+    )
+
+    region = module.params.get('region')
+    aws_secret_key = module.params.get('aws_secret_key')
+    aws_access_key = module.params.get('aws_access_key')
+
+    if not aws_secret_key:
+        if  'AWS_SECRET_KEY' in os.environ:
+            aws_secret_key = os.environ['AWS_SECRET_KEY']
+        elif 'EC2_SECRET_KEY' in os.environ:
+            aws_secret_key = os.environ['EC2_SECRET_KEY']
+
+    if not aws_access_key:
+        if 'AWS_ACCESS_KEY' in os.environ:
+            aws_access_key = os.environ['AWS_ACCESS_KEY']
+        elif 'EC2_ACCESS_KEY' in os.environ:
+            aws_access_key = os.environ['EC2_ACCESS_KEY']
+
```

Comment:
Functions for reading AWS credentials have been standardized in the module ansible.module_utils.ec2.  To be consistent with the other AWS modules you should import that module and use the function get_aws_connection_info to fetch AWS_ACCESS_KEY, AWS_SECRET_KEY, etc. rather than doing it yourself.  See, for example, the [ec2 module](https://github.com/ansible/ansible-modules-core/blob/devel/cloud/amazon/ec2.py#L1226).